openjpa-kernel/src/main/java/org/apache/openjpa/meta/JavaTypes.java

```diff
@@ -417,6 +421,11 @@ public static Object convert(Object val, int typeCode) {
                 if (val instanceof String)
                     return Short.valueOf(val.toString());
                 return val;
+            case UUID_OBJ:
+                if (val instanceof String) {
+                    return UUID.fromString(val.toString());
```

Comment:
```suggestion
                    return UUID.fromString((String)val);
```
?

Reply:
I used ```val.toString()``` because of how other cases use it (such as LONG, INT etc.). Exchanged for (String) val as suggested and tested again on hsql, h2, h2-2, pg-13 to pg-17, mysql lts and latest, mariadb lts and latest, and oracle XE 21c.